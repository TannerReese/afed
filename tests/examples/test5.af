# Test vector and matrix arithmetic

vec1: V[1, 2, 3],
vec2: V[3, -2, 5],
vec3: V[10, -4.2, 1, 1 + 2 / 3],
vec4: V[3, -3, 3, 2],

vec4 + vec1 = ``,
vec1 - vec3 = ``,
vec3 * vec2 = ``,
V[2 * 4, true + "Hello", 1 - 3] = ``,

2.0 * (vec1 - vec2 / 3) = ``,
vec3 + (vec1 * vec2) * vec4 = ``,
vec1 * vec2 + vec3 * vec4 = ``,
vec1 1 + vec2 2 + vec3 0 + vec3 3 = ``,
vec.mag vec1 = ``,
[vec1, vec1] == [vec1 / 5 * 5, vec1 - vec2 * 10 + vec2 * 10] = ``,
vec.dims vec4 = ``,

