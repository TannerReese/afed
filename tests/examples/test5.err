line 8, column 15 Eval Error: Vector dimensions 4 and 3 do not match
line 9, column 15 Eval Error: Vector dimensions 3 and 4 do not match
line 10, column 15 Eval Error: Vector dimensions 4 and 3 do not match
line 11, column 35 Eval Error: Binary operator + not implemented between types boolean and string
4 Eval Errors encountered
