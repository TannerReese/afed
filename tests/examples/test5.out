# Test vector and matrix arithmetic

vec1: V[1, 2, 3],
vec2: V[3, -2, 5],
vec3: V[10, -4.2, 1, 1 + 2 / 3],
vec4: V[3, -3, 3, 2],

vec4 + vec1 = `Eval Error: Vector dimensions 4 and 3 do not match`,
vec1 - vec3 = `Eval Error: Vector dimensions 3 and 4 do not match`,
vec3 * vec2 = `Eval Error: Vector dimensions 4 and 3 do not match`,
V[2 * 4, true + "Hello", 1 - 3] = `Eval Error: Binary operator + not implemented between types boolean and string`,

2.0 * (vec1 - vec2 / 3) = `V[0, 5.333333333333333, 2.6666666666666665]`,
vec3 + (vec1 * vec2) * vec4 = `V[52, -46.2, 43, 89 / 3]`,
vec1 * vec2 + vec3 * vec4 = `62.93333333333334`,
vec1 1 + vec2 2 + vec3 0 + vec3 3 = `56 / 3`,
vec.mag vec1 = `3.7416573867739413`,
[vec1, vec1] == [vec1 / 5 * 5, vec1 - vec2 * 10 + vec2 * 10] = `true`,
vec.dims vec4 = `4`,

